src/core/handle.ml

```diff
@@ -22,12 +22,10 @@ let check_builtin_nat : popt -> sym StrMap.t -> string -> sym -> unit
   match s with
   | "+1" ->
      let builtin = Sign.builtin pos builtins in
-     let symb_0 = builtin "0" in
-     let typ_0 = !(symb_0.sym_type) in
-     let x = Bindlib.new_var mkfree "_" in
-     let typ_s = Ctxt.to_prod [(x, typ_0)] typ_0 in
-     if not (Basics.eq typ_s !(sym.sym_type)) then
-       fatal pos "The type of [%s] is not of the form [%a]"
+     let typ_0 = !((builtin "0").sym_type) in
+     let typ_s = let y = lift typ_0 in Bindlib.unbox (_Impl y y) in
```

Comment:
`y` is not a great name here, why not use `typ_0`?

Reply:
Well, why not writing `let typ_0 = lift !((builtin "0").sym_type) in let typ_s = Bindlib.unbox (_Impl typ_0 typ_0) in` as I suggested in a previous comment? ;-)